compute_sdk/tests/unit/test_web_client.py

```diff
@@ -139,3 +152,30 @@ def test_delete_function(client: WebClient, randomstring: t.Callable):
     res = client.delete_function(func_uuid_str)
 
     assert res["some_key"] == expected_response
+
+
+@pytest.mark.parametrize("ep_uuid", [uuid.uuid4(), None])
+def test_register_endpoint_post_put(
+    client: WebClient, randomstring: t.Callable, ep_uuid: t.Optional[uuid.UUID]
+):
+    post_response = {"foo": randomstring()}
+    put_response = {"foo": randomstring()}
+    if ep_uuid:
+        responses.add(
+            responses.PUT,
+            f"https://api.funcx/v3/endpoints/{ep_uuid}",
+            json=put_response,
+        )
+    else:
+        responses.add(
+            responses.POST,
+            "https://api.funcx/v3/endpoints",
+            json=post_response,
```

Comment:
If it makes it cleaner (I haven't tried it), I observe that we're using the same prefix (`https://api.funcx/v3/endpoints`) in a few places in this file.  A file-level variable might reduce copy-pasta:

```python
_V3BASE = "https://api.gc/v3/endpoints"
...
    if ep_uuid:
        responses.put(f"{_V3BASE}/{ep_uuid}", json=put_response)
    else:
        responses.post(_V3BASE, json=post_response)
...
```

Reply:
`put_response` and `post_response` are the same -- so, redundant.  Could simplify test:
```python
    res_data = {"foo": randomstring()}
    if ep_uuid:
        responses.put(f"{_V3BASE}/{ep_uuid}", json=res_data)
    else:
        responses.post(_V3BASE, json=res_data)
    res = client.register_endpoint("MyEP", ep_uuid)

    assert res.data == res_data
```

Reply:
I updated the test to clearly differentiate between a failure caused by the `WebClient` using the incorrect method and URL mismatch or typo.